src/BloomExe/Command.cs

```diff
@@ -12,43 +12,95 @@ public interface ICommand
 	{
 		void Execute();
 		bool Enabled { get; set; }
-		//event EventHandler EnabledChanged;
+		event EventHandler EnabledChanged;
 	}
 
 	public abstract class Command : ICommand
 	{
-		//public event EventHandler EnabledChanged;
-		public bool Enabled { get; set; }
-		public Command()
+		public event EventHandler EnabledChanged;
+		private bool _enabled;
+
+		public Command(string name)
 		{
-			Enabled = true;
+			Name = name;
+			_enabled = true;
 		}
 
-//		public bool Enabled
-//		{
-//			get {return Implementer !=null && _enabled; }
-//			private set
-//			{
-//				_enabled = value;
-//				if (EnabledChanged != null)
-//				{
-//					EnabledChanged.Invoke(this, null);
-//				}
-//			}
-//        }
-		public Action Implementer { get; set;}
+		public bool Enabled
+		{
+			get { return Implementer != null && _enabled; }
+			set
+			{
+				_enabled = value;
+				RaiseEnabledChanged();
+			}
+		}
+		public Action Implementer { get; set; }
+		public string Name { get; set; }
 
 		public void Execute()
 		{
 			Implementer();
 		}
+		protected virtual void RaiseEnabledChanged()
+		{
+			var handler = EnabledChanged;
+			if(handler != null)
+				handler(this, EventArgs.Empty);
+		}
+	}
+
+	public class CutCommand : Command
+	{
+		public CutCommand()
+			: base("cut")
+		{
+
+		}
+	}
+
+	public class CopyCommand : Command
+	{
+		public CopyCommand()
+			: base("copy")
+		{
+
+		}
 	}
 
-	public class CutCommand : Command { }
-	public class CopyCommand : Command { }
-	public class PasteCommand : Command{}
-	public class UndoCommand : Command { }
-	public class DuplicatePageCommand : Command { }
-	public class DeletePageCommand : Command { }
+	public class PasteCommand : Command
```

Comment:
It feels vaguely heavy-weight to have a whole class for each command, if they only differ by the value of one field. But there may be something about the framework that requires this.